torch/utils/throughput_benchmark.py

```diff
@@ -2,6 +2,61 @@
 
 import torch._C
 
+def format_time(time_us=None, time_ms=None, time_s=None):
+    '''Defines how to format time'''
+    assert sum([time_us is not None, time_ms is not None, time_s is not None]) == 1
+    if time_us is None:
+        if time_ms is not None:
+            time_us = time_ms * 1000.0
+        elif time_s is not None:
+            time_us = time_s * 1000.0 * 1000.0
+        else:
+            assert "Incorrect code :)"
+
+    US_IN_SECOND = 1000.0 * 1000.0
+    US_IN_MS = 1000.0
+    if time_us >= US_IN_SECOND:
+        return '{:.3f}s'.format(time_us / US_IN_SECOND)
+    if time_us >= US_IN_MS:
+        return '{:.3f}ms'.format(time_us / US_IN_MS)
+    return '{:.3f}us'.format(time_us)
+
+
+class ExecutionStats(object):
+    def __init__(self, c_stats, benchmark_config):
+        self._c_stats = c_stats
+        self.benchmark_config = benchmark_config
+
+    @property
+    def latency_avg_ms(self):
+        return self._c_stats.latency_avg_ms
+
+    @property
+    def num_iters(self):
+        return self._c_stats.num_iters
+
+    @property
+    def iters_per_second(self):
+        '''
+        Returns total number of iterations per second across all calling threads
+        '''
+        return self.num_iters / self.total_time_seconds
+
+    @property
+    def total_time_seconds(self):
```

Comment:
alternative approach would have been to just assign these methods on the class, e.g.:

```
def total_time_seconds(self):
  ...

torch._C.ExecutionStats.total_time_seconds = property(total_time_seconds)
```

That way you don't need to create a wrapper class. But I don't insist - either works.

Reply:
Hm, that would be quite a bit of injection of extra methods, including __str__. I personally find it a bit more readable with an explicit wrapper class.